src/OrderImport/Rules/Cdiscount.php

```diff
@@ -199,4 +89,25 @@ public function getDescription()
     {
         return $this->l('Adds an \'Operation Fee\' product to the order, so the amount will show in the invoice.', 'Cdiscount');
     }
+
+    protected function getReference()
+    {
+        $product = $this->initCdiscountFeeProduct()->getProduct();
+        $referenceFormat = Configuration::get(Shoppingfeed::PRODUCT_FEED_REFERENCE_FORMAT);
```

Comment:
To prevent BC, you should $product->getReference(); 
See also my comment on this method. A merchand that updagrade sould have the same item reference => 'FDG-ShoppingFlux'.